src/components/LoginComponent.js

```diff
@@ -0,0 +1,195 @@
+import React, { useReducer } from 'react';
+import {
+  TextField,
+  Button,
+  makeStyles,
+  InputAdornment,
+  IconButton,
+} from '@material-ui/core';
+import { Visibility, VisibilityOff, Email } from '@material-ui/icons';
+import '../App.css';
+import { registerUser, signInUser } from '../utils/userAuth';
+import { useHistory, Redirect } from 'react-router-dom';
+
+const useStyles = makeStyles({
+  inputContainer: {
+    marginBottom: 20,
+    width: '80%',
+    textAlign: 'center',
+  },
+  textField: {
+    color: 'white',
+  },
+  inputLabel: {
+    color: '#aaa',
+  },
+  icon: {
+    color: 'white',
+    marginRight: 10,
+    padding: 0,
+  },
+  iconContainer: {
+    marginRight: 10,
+  },
+  zeroMargin: {
+    margin: 0,
+  },
+});
+
+const LoginComponent = ({ notifyLogIn, displayPopup, loggedIn, route }) => {
+  const classes = useStyles();
+  const history = useHistory();
+
+  const initialState = {
+    email: '',
+    pass: '',
+    showPass: false,
+    passTwo: '',
+    login: true,
+  };
+  const [state, setState] = useReducer(
+    (state, newState) => ({ ...state, ...newState }),
+    initialState
+  );
+
+  const handleRegisterButton = async () => {
+    try {
+      const result = await (state.login
+        ? signInUser(state.email, state.pass)
+        : registerUser(state.email, state.pass));
+      notifyLogIn();
+      sessionStorage.setItem('user', JSON.stringify(result));
+      history.push('/home');
```

Comment:
Just curious as to what this does

Reply:
Id assume push them to the homepage once they have logged in?

Reply:
yeah it's a weirdly named way of redirecting to home